src/Test/test_conic.jl

```diff
@@ -6831,3 +6831,206 @@ function setup_test(
+    @test ≈(MOI.Utilities.set_dot(soc_primal, soc_dual, soc_set), T(0), config)
+    @test ≈(MOI.get(model, MOI.VariablePrimal(), x), primal, config)
+    @test ≈(MOI.get(model, MOI.VariablePrimal(), t), t_value, config)
+    @test ≈(MOI.get(model, MOI.ConstraintPrimal(), cx), primal, config)
+    @test ≈(MOI.get(model, MOI.ConstraintPrimal(), con_soc), soc_primal, config)
+    if _supports(config, MOI.ConstraintDual)
+        @test ≈(MOI.get(model, MOI.ConstraintDual(), cx), dual, config)
+        @test ≈(MOI.get(model, MOI.ConstraintDual(), con_soc), soc_dual, config)
+    end
+    return
+end
+
+function setup_test(
+    ::typeof(test_conic_HermitianPositiveSemidefiniteConeTriangle_1),
+    model::MOIU.MockOptimizer,
+    ::Config{T},
+) where {T}
+    primal = [
+        (T(1) + √T(3)) / T(2),
+        -T(1) / T(2),
+        (-T(1) + √T(3)) / T(2),
+        T(1) / T(2),
+    ]
+    soc_primal = [
+        primal[1] - one(T),
+        √T(2) * (primal[2] + one(T)),
+        primal[3] + one(T),
+        √T(2) * (primal[4] - one(T)),
+    ]
+    t_value = LinearAlgebra.norm(soc_primal)
+    push!(primal, t_value)
+    soc_primal = [t_value; soc_primal]
+    dual = [
+        (T(3) - √T(3)) / T(6),
+        √T(3) / T(6),
+        (T(3) + √T(3)) / T(6),
+        -√T(3) / T(6),
+    ]
+    soc_dual = [one(T), -dual[1], -dual[2] * √T(2), -dual[3], -dual[4] * √T(2)]
+    MOIU.set_mock_optimize!(
+        model,
+        (mock::MOIU.MockOptimizer) -> MOIU.mock_optimize!(
+            mock,
+            primal,
+            (MOI.VectorAffineFunction{T}, MOI.SecondOrderCone) =>
+                [soc_dual],
+        ),
+    )
+    return
+end
+
+function version_added(
+    ::typeof(test_conic_HermitianPositiveSemidefiniteConeTriangle_1),
+)
+    return v"1.7.0"
+end
+
+"""
+    test_conic_HermitianPositiveSemidefiniteConeTriangle_2(
+        model::MOI.ModelLike,
+        config::Config,
+    )
+
+Test adding [`MathOptInterface.VariableIndex`](@ref)-in-[`MathOptInterface.EqualTo](@ref)`
+on [`MathOptInterface.HermitianPositiveSemidefiniteConeTriangle`](@ref)
+variables.
+```
+"""
+function test_conic_HermitianPositiveSemidefiniteConeTriangle_2(
+    model::MOI.ModelLike,
+    config::Config{T},
+) where {T}
+    @requires _supports(config, MOI.optimize!)
+    @requires MOI.supports_add_constrained_variables(
+        model,
+        MOI.HermitianPositiveSemidefiniteConeTriangle,
+    )
+    @requires MOI.supports_constraint(model, MOI.VariableIndex, MOI.EqualTo{T})
+    set = MOI.HermitianPositiveSemidefiniteConeTriangle(3)
+    x, cx = MOI.add_constrained_variables(model, set)
+    primal = T[1, 0, 1, 0, 0, 0, -1, 0, 0]
+    MOI.add_constraints(model, x, MOI.EqualTo.(primal))
+    MOI.optimize!(model)
+    @test ≈(MOI.get(model, MOI.VariablePrimal(), x), primal, config)
+    @test ≈(MOI.get(model, MOI.ConstraintPrimal(), cx), primal, config)
+    if _supports(config, MOI.ConstraintDual)
+        dual = zeros(T, 9)
```

Comment:
@blegat, some solvers are failing this test: https://github.com/jump-dev/SolverTests/actions/runs/2857418874

Is it possible for the dual to be non-zero?